src/mbgl/vulkan/context.cpp

```diff
@@ -173,18 +197,26 @@ void Context::beginFrame() {
     device->resetDescriptorPool(getCurrentDescriptorPool().get());
     frame.runDeletionQueue(*this);
 
-    if (renderableResource.getPlatformSurface()) {
+    if (platformSurface) {
         try {
             const vk::ResultValue acquireImageResult = device->acquireNextImageKHR(
                 renderableResource.getSwapchain().get(), timeout, frame.surfaceSemaphore.get(), nullptr);
 
-            if (acquireImageResult.result == vk::Result::eSuccess)
+            if (acquireImageResult.result == vk::Result::eSuccess) {
+                renderableResource.setAcquiredImageIndex(acquireImageResult.value);
+            } else if (acquireImageResult.result == vk::Result::eSuboptimalKHR) {
                 renderableResource.setAcquiredImageIndex(acquireImageResult.value);
-            else if (acquireImageResult.result == vk::Result::eSuboptimalKHR)
-                renderableResource.recreateSwapchain();
+                // TODO implement pre-rotation transform for surface orientation
```

Comment:
Worth creating an issue for?

Reply:
Technically it's an optimization. I might add an issue with a checklist for all planned improvements